src/python/pants/engine/target.py

```diff
@@ -351,9 +351,18 @@ def _calculate_field_values(
             valid_aliases.add(field_type.alias)
             aliases_to_field_types[field_type.alias] = field_type
             if field_type.deprecated_alias is not None:
+                valid_aliases.add(field_type.deprecated_alias)
```

Comment:
Note that this is new. I think it was an oversight, we should have always done this.